### ISSUE TITLE
options to configure locations of linked applications (on darwin)

### ISSUE THREAD

**Issue by sellout** [nix-community/home-manager#3557]
I like having the darwin applications symlinked to ~/Applications/Home Manager Apps, however there is at least one application that refuses to run from there. 1Password _must_ be in either /Applications or ~/Applications _directly_, not in a subdirectory.
<img width="372" alt="Screenshot 2023-01-02 at 15 34 53" src="https://user-images.githubusercontent.com/33031/210282306-0719af20-912e-4924-8219-1f5c6c0bc7f4.png">
Maybe something like
```nix
{
  targets.darwin.applications = {
    enable = true;
    defaultPath = "${config.home.homeDirectory}/Applications/Home Manager Apps";
    customPaths = {
      # NB: 1Password refuses to run if it’s not installed here.
      "${pkgs._1password-gui}" = "${config.home.homeDirectory}/Applications";
    };
}
```
or `targets.darwin.linkedApplications`?

**Comment by seh** [comment]
@sellout, what is that "targets.darwin.applications" attribute defined? Does it exist today? If so, what reads it?

**Comment by sellout** [comment]
@seh No, it doesn’t exist. `targets.darwin` exists for darwin-specific attributes, so I’m proposing adding `applications` (or perhaps `linkedApplications`) to support this feature.

[nix-darwin](https://github.com/LnL7/nix-darwin) actually does something similar in [its homebrew module](https://github.com/LnL7/nix-darwin/blob/master/modules/homebrew.nix), so that might also be somethnig to take inspiration from – but it really just exposes the Brewfile options.
